hnclient-v2/src/main/java/ca/bc/gov/hlth/hnclientv2/handshakeserver/ConnectionHandler.java

```diff
@@ -190,25 +190,22 @@ private void performTransaction(BufferedInputStream socketInput, BufferedOutputS
 
 		logger.info("{} - TransactionId: {} Start performing message transaction: {}", methodName, transactionId,  ret_code);
 		// read SI segment
-		if (socketInput.available() > 0) {
-			byte[] message = new byte[SI_MESSAGE_LENGTH];
-			socketInput.read(message);
-			
-			HandshakeUtil.unScrambleData(message, decodeSeed);
-			logInputData(message);					
-			logger.debug("{} - TransactionId: {}  Received from originator {} byte SI Data Block: {}", methodName, transactionId,  message.length,
+		
+		logger.info("{} - TransactionId: {} Start reading SI segment: {}", methodName, transactionId,  ret_code);
+		byte[] sisegment = new byte[SI_MESSAGE_LENGTH];
+		socketInput.read(sisegment);			
+		HandshakeUtil.unScrambleData(sisegment, decodeSeed);
+		logInputData(sisegment);					
+		logger.debug("{} - TransactionId: {}  Received from originator {} byte SI Data Block: {}", methodName, transactionId,  sisegment.length,
 					ret_code);
-			logger.debug("{} - TransactionId: {}  Received SI segment: {}", methodName, transactionId,  new String(message, StandardCharsets.UTF_8));
-		} else {
-			ret_code = MessageUtil.HNET_RTRN_INVALIDFORMATERROR;
-			logger.debug("{} - TransactionId: {}  Error receiving SI segment from Listener: {}", methodName, transactionId, ret_code);
-
-		}
-
+		logger.debug("{} - TransactionId: {}  Received SI segment: {}", methodName, transactionId,  new String(sisegment, StandardCharsets.UTF_8));
+		
 		// read dtsegment header
+		logger.info("{} - TransactionId: {} Start reading DT segment: {}", methodName, transactionId,  ret_code);
```

Comment:
Do we want info level logging here? I suspect at some in Prod, once things have stabilized, we might want to change the logging level from debug to info to not include these kind of detailed logs. It's really up to you though. Same comment applies below.